pkg/cloudprovider/aws/instancetype.go

```diff
@@ -0,0 +1,115 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/awslabs/karpenter/pkg/cloudprovider/aws/utils"
+	"github.com/awslabs/karpenter/pkg/utils/resources"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type InstanceType struct {
+	ec2.InstanceTypeInfo
+	ZoneOptions []string
+}
+
+func (i *InstanceType) Name() string {
+	return aws.StringValue(i.InstanceType)
+}
+func (i *InstanceType) Zones() []string {
+	return i.ZoneOptions
+}
+
+func (i *InstanceType) Architectures() []string {
+	architectures := []string{}
+	for _, architecture := range i.ProcessorInfo.SupportedArchitectures {
+		architectures = append(architectures, utils.KubeArchitecture(aws.StringValue(architecture)))
+	}
+	return architectures
+}
+
+func (i *InstanceType) OperatingSystems() []string {
+	return SupportedOperatingSystems
+}
+
+func (i *InstanceType) CPU() *resource.Quantity {
+	return resources.Quantity(fmt.Sprint(*i.VCpuInfo.DefaultVCpus))
+}
+
+func (i *InstanceType) Memory() *resource.Quantity {
+	return resources.Quantity(fmt.Sprintf("%dMi", *i.MemoryInfo.SizeInMiB))
+}
+
+func (i *InstanceType) Pods() *resource.Quantity {
+	// The number of pods per node is calculated using the formula:
+	// max number of ENIs * (IPv4 Addresses per ENI -1) + 2
+	// https://github.com/awslabs/amazon-eks-ami/blob/master/files/eni-max-pods.txt#L20
+	return resources.Quantity(fmt.Sprint(*i.NetworkInfo.MaximumNetworkInterfaces*(*i.NetworkInfo.Ipv4AddressesPerInterface-1) + 2))
+}
+
+func (i *InstanceType) NvidiaGPUs() *resource.Quantity {
+	count := int64(0)
+	if i.GpuInfo != nil {
+		for _, gpu := range i.GpuInfo.Gpus {
+			if *i.GpuInfo.Gpus[0].Manufacturer == "NVIDIA" {
+				count += *gpu.Count
+			}
+		}
+	}
+	return resources.Quantity(fmt.Sprint(count))
+}
+
+func (i *InstanceType) AWSNeurons() *resource.Quantity {
+	count := int64(0)
+	if i.InferenceAcceleratorInfo != nil {
+		for _, accelerator := range i.InferenceAcceleratorInfo.Accelerators {
+			count += *accelerator.Count
```

Comment:
I don't follow what neurons means in this context - how are "accelerators" used in instance type selection? Is this  basically number of inferentia chips?

Reply:
You got it
```
resources:
  requests:
     amazonaws.com/neuron: 3 # get me a machine w/ 3 chips
```